app/frontend/actions.ts

```diff
@@ -564,63 +583,39 @@ export default ({setState, getState}: {setState: SetStateFn; getState: GetStateF
     setErrorState('sendAddressValidationError', sendAddressValidator(state.sendAddress.fieldValue))
     setErrorState(
       'sendAmountValidationError',
-      sendAmountValidator(
-        state.sendAmount.fieldValue,
-        state.sendAmount.coins,
-        getSourceAccountInfo(state).balance
-      )
+      sendAmountValidator(state.sendAmount.fieldValue, state.sendAmount.coins, getSourceAccountInfo(
+        state
+      ).balance as Lovelace)
     )
     setErrorState(
       'donationAmountValidationError',
       donationAmountValidator(
         state.donationAmount.fieldValue,
         state.donationAmount.coins,
-        getSourceAccountInfo(state).balance
+        getSourceAccountInfo(state).balance as Lovelace
       )
     )
   }
 
-  const resetDelegationWithoutHash = (state) => {
-    const newState = getState()
-    setState({
-      shelleyDelegation: {
-        ...state.shelleyDelegation,
-        selectedPool: {
-          ...newState.shelleyDelegation.selectedPool,
-        },
-        delegationFee: 0,
-      },
-    })
-  }
-
   const isSendFormFilledAndValid = (state) =>
     state.sendAddress.fieldValue !== '' &&
     state.sendAmount.fieldValue !== '' &&
     !state.sendAddressValidationError &&
     !state.sendAmountValidationError &&
     !state.donationAmountValidationError
 
-  const prepareTxPlan = async (args) => {
+  const prepareTxPlan = async (args: TxPlanArgs): Promise<TxPlanResult> => {
     const state = getState()
     try {
-      const plan: any = await wallet.getAccount(state.sourceAccountIndex).getTxPlan(args)
-      // FIXME: this cant be any
-      if (plan.error) {
-        stopLoadingAction(state, {})
-        resetDelegationWithoutHash(state)
-        setState({
-          calculatingDelegationFee: false,
-          calculatingFee: false,
-        })
-      }
-      return plan
+      return await wallet.getAccount(state.sourceAccountIndex).getTxPlan(args)
```

Comment:
For the future: I hate that this contains side-effects. In reality, we should split out the functionality into fetching of available inputs and "coin selection" algorithm